Source/WebCore/history/BackForwardController.cpp

```diff
@@ -170,6 +170,17 @@ RefPtr<HistoryItem> BackForwardController::itemAtIndex(int i)
     return protectedClient()->itemAtIndex(i);
 }
 
+Vector<Ref<HistoryItem>> BackForwardController::allItems()
+{
+    Vector<Ref<HistoryItem>> historyItems;
+    for (int index = -1 * static_cast<int>(backCount()); index <= static_cast<int>(forwardCount()); index++) {
+        if (RefPtr item = itemAtIndex(index))
```

Comment:
else ASSERT_NOT_REACHED()

Reply:
I've hit this condition before.

My assumption was that `WebBackForwardListProxy` is to another process so there is a possible race condition between getting the size and an item.

Reply:
I guess the better solution here is a new API that returns a whole list from the client.